test/e2e/util/scheduler_injector.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2025 The Volcano Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"context"
+
+	v1core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// schedulerInjectingClientset is used to wrap the Kubernetes clientset and inject a specific scheduler name, which is volcano in our e2e cases by default.
+// Then we don't need to use like gomonkey tools to patch k8s e2e methods to inject the scheduler name.
+type schedulerInjectingClientset struct {
+	kubernetes.Interface
+	schedulerName string
+}
+
+func (c *schedulerInjectingClientset) CoreV1() v1.CoreV1Interface {
+	return &schedulerInjectingCoreV1{
+		CoreV1Interface: c.Interface.CoreV1(),
+		schedulerName:   c.schedulerName,
+	}
+}
+
+// schedulerInjectingCoreV1 wraps CoreV1Interface and intercepts the Pods method.
+type schedulerInjectingCoreV1 struct {
+	v1.CoreV1Interface
+	schedulerName string
+}
+
+func (c *schedulerInjectingCoreV1) Pods(namespace string) v1.PodInterface {
+	return &schedulerInjectingPodInterface{
+		PodInterface:  c.CoreV1Interface.Pods(namespace),
+		schedulerName: c.schedulerName,
+	}
+}
+
+// schedulerInjectingPodInterface wraps PodInterface and intercepts the Create method to inject a scheduler name into the Pod spec.
+type schedulerInjectingPodInterface struct {
+	v1.PodInterface
+	schedulerName string
+}
+
+func (p *schedulerInjectingPodInterface) Create(ctx context.Context, pod *v1core.Pod, opts metav1.CreateOptions) (*v1core.Pod, error) {
+	if pod.Spec.SchedulerName == "" {
```

Comment:
What if pods already set a schedulerName?

Reply:
Then I think these pods should be scheduled by other schedulers, such as kube-scheduler, but in our use cases there are no cases specify schedulerName to default-scheduler, it should be fine. And currently scheduler_injector is only used in volume_binding e2e testing cases

Reply:
No in volume binding e2e testing, the pod functions related to creation (copied from the kubernetes repo) do not set the schedulerName. My injector is to inject the schedulerName before issuing the pod creation request. If the schedulerName is set to volcano, the kube-apiserver will not set the default value for this pod.

Reply:
We can't determine the behavior of k8s e2e, so it's better to rewrite this field directly.

Reply:
OK. Updated